src/DynamoCoreWpf/ViewModels/RunSettingsViewModel.cs

```diff
@@ -256,6 +256,16 @@ void Model_PropertyChanged(object sender, System.ComponentModel.PropertyChangedE
                     RaisePropertyChanged("RunEnabled");
                     RaisePropertyChanged("RunButtonEnabled");
                     RaisePropertyChanged("RunButtonToolTip");
+                    if (Application.Current != null)
+                    {
+                        Application.Current.Dispatcher.Invoke(new Action(() =>
+                        {
+                            dynamoViewModel.ShowOpenDialogAndOpenResultCommand.RaiseCanExecuteChanged();
+                            dynamoViewModel.NewHomeWorkspaceCommand.RaiseCanExecuteChanged();
+                            dynamoViewModel.OpenRecentCommand.RaiseCanExecuteChanged();
+                            dynamoViewModel.CloseHomeWorkspaceCommand.RaiseCanExecuteChanged();
+                        }));
+                    }
```

Comment:
Are we saying that `Model_PropertyChanged` might not be called on the main thread?